### PR TITLE
Fix out-of-bounds error within Gitea integration

### PR DESCRIPTION
**Pull request by tboerger**
Port from https://github.com/drone/drone/pull/2434

### CODE REVIEW

File: scm/driver/gitea/webhook.go [hunk 1]
```diff
@@ -190,27 +190,41 @@ func convertBranchHook(dst *createHook, action scm.Action) *scm.BranchHook {
...
+			Repo:   *convertRepository(&dst.Repository),
+			Sender: *convertUser(&dst.Sender),
+		}
+	} else {
+		return &scm.PushHook{
+			Ref: dst.Ref,
+			Commit: scm.Commit{
+				Sha:       dst.After,
+				Link:      dst.Compare,
+				Author:    scm.Signature{},
```
Comment (bradrydzewski) [inline scm/driver/gitea/webhook.go]: maybe we use the `pusher` value here? https://github.com/drone/go-scm/blob/master/scm/driver/gitea/testdata/webhooks/push.json#L64:L70

Comment (tboerger) [reply scm/driver/gitea/webhook.go]: For author and commiter? Or only for one of them?

Comment (bradrydzewski) [reply scm/driver/gitea/webhook.go]: I'm thinking for both ...

**Comment by tboerger** [comment]
I have updated it with the values from `Pusher`

**Pull request merged**